Add fuzzing test for `yaml.Marshal` and `yaml.Unmarshal`

rhysd opened the pull request:
This PR adds [a fuzzing test](https://en.wikipedia.org/wiki/Fuzzing) for go-yaml package using [Go's standard fuzzing framework](https://go.dev/doc/security/fuzz/).

- Add a minimal fuzz test case to validate `yaml.Unmarshal` and `yaml.Marshal` work without panics
- Add `make` task to run fuzzing test in one command (`make fuzz`)
- Add CI workflow which runs the fuzzing test for 10 minutes weekly (on every Saturday 6:00 AM). Since running fuzzing test takes a bit longer time, it is not suitable for running on every commit

ccoVeille commented on GNUmakefile:
```diff
@@ -39,6 +39,7 @@ include $(MAKES)/shell.mk
 
 v ?=
 count ?= 1
+time ?= 60s
```
I would fuzztime as a variable name then

ccoVeille commented on fuzz_test.go:
```diff
@@ -0,0 +1,43 @@
+	"go.yaml.in/yaml/v4"
+)
+
+func setupSeedCorpus(f *testing.F) {
```
```suggestion
func setupSeedCorpus(f *testing.F) {
	f.Helper()

```

SVilgelm commented on .github/workflows/fuzz.yaml:
```diff
@@ -0,0 +1,25 @@
+  - cron: 0 6 * * 6
+
+permissions:
+  contents: read  # for actions/checkout to fetch code
```
please change it to `read-all`, because it is unclear what permissions are in use for other types

```suggestion
permissions: read-all
```

rhysd commented:
I apologize for the lack of response because I was working on other projects. I rebased this branch onto the latest main branch and addressed all comments.

ingydotnet commented:
See #210

nikolaydubina commented:
interesting. probably effective fuzzing can be done even without LibAFL. you just need effective corpus. probably can run AI loop: read whole repo -> run fuzz tests -> get coverage -> feed that back to LLM -> ask LLM to write new test input file -> add it to `/testdata` in repo -> repeat.

probably this can be even done in some simple bash script or something and you will really rapidly accumulate a very good test corpus. no need to waste 30s of CPU cycles either on every push haha.

Happy New year guys! 🎄
